plutus-core/executables/uplc/Main.hs

```diff
@@ -268,64 +293,72 @@ runApplyToData (ApplyOptions inputfiles ifmt outp ofmt mode) =
                        Right (d :: Data) ->
                            pure $ UPLC.Program () ver $ mkConstant () d
 
+---------------- Benchmarking ----------------
+
+runBenchmark :: BenchmarkOptions -> IO ()
+runBenchmark (BenchmarkOptions inp ifmt semvar timeLim) = do
+  prog <- readProgram ifmt inp
+  let criterionConfig = defaultConfig {reportFile = Nothing, timeLimit = timeLim}
+      cekparams = mkMachineParameters semvar PLC.defaultCekCostModel
+      getResult (x,_,_) = either (error . show) (\_ -> ()) x  -- Extract an evaluation result
+      evaluate = getResult . Cek.runCekDeBruijn cekparams Cek.restrictingEnormous Cek.noEmitter
+      -- readProgam throws away De Bruijn indices and returns an AST with Names;
+      -- we have to put them back to get an AST with NamedDeBruijn names.
+      !term = fromRight (error "Unexpected open term in runBenchmark.") .
+                runExcept @FreeVariableError $ UPLC.deBruijnTerm (UPLC._progTerm prog)
+      -- Big names slow things down
+      !anonTerm = UPLC.termMapNames (\(PLC.NamedDeBruijn _ i) -> PLC.NamedDeBruijn "" i) term
+      -- Big annotations slow things down
+      !unitAnnTerm = force (() <$ anonTerm)
+  benchmarkWith criterionConfig $! whnf evaluate unitAnnTerm
```

Comment:
Would it be possible to reuse `Common.benchWith` from the benchmarks to make sure it's not how we run the benchmarks that makes the difference?